### ISSUE TITLE
The inference time of the int8 exported ONNX model is abnormal.

### ISSUE THREAD

**Issue by Xinyu2Sy** [ultralytics/ultralytics#18109]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I use model.export() to convert yolov11n.pt to ONNX, I found that the inference time of the model with int8 precision is not significantly different from the model without int8 precision (based on the average inference time of n runs in the same environment).

Average inference time   (Inference count of 100)
yolov11n.onnx with int8 true: 32.57 ms
yolov11n.onnx with int8 False: 22.34 ms

Average inference time   (Inference count of 1000)
yolov11n.onnx with int8 true: 22.38 ms
yolov11n.onnx with int8 False: 22.34 ms

Is this phenomenon normal? If not, could you provide some troubleshooting ideas?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Xinyu2Sy, thank you for bringing up your concern about ONNX inference times 🚀!

We suggest starting with our [Docs](https://docs.ultralytics.com), where you can find resources on exporting models and optimizing inference. Specifically, the [export section](https://docs.ultralytics.com/modes/export/) has details on using `model.export()` with different formats and quantization settings.

If this is a 🐛 Bug Report, could you kindly provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/)? This would allow us to investigate the issue more effectively. For example, share a snippet of the code you are using for the export and timing process, along with details about your hardware and software environment (e.g., OS, CUDA, PyTorch, ONNX version, etc.).

If you're troubleshooting custom setups ❓, some guidelines to ensure proper benchmarking include:
- Verifying that your ONNX runtime or inference engine supports int8 quantization correctly.
- Ensuring you're running on appropriate hardware (e.g., GPU/CPU) optimized for int8 precision.
- Testing with a clean environment and the latest `ultralytics` package to rule out version-related issues.

Join our Ultralytics community for additional insights! For real-time conversation, hop on our [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth asynchronous discussions, check out [Discourse](https://community.ultralytics.com). Or participate in knowledge sharing on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Please confirm you're using the latest version of the `ultralytics` package with all requirements. You can upgrade by running:

```bash
pip install -U ultralytics
```

This ensures you're benefiting from the latest fixes and enhancements.

## Environments

For accurate benchmarking and exports, we recommend verified runtime environments. YOLO can run in several pre-configured platforms:
- **Notebooks** with free GPU access: 
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
- Refer to our [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) guides for other options.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, verifying operational correctness across all supported [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

This is an automated response to guide you. Please rest assured, an Ultralytics engineer will review your issue and provide further assistance soon ✨.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/17158

**Comment by Xinyu2Sy** [comment]
> Duplicate of #17158

Thanks for your feedback!

**Comment by glenn-jocher** [comment]
@Xinyu2Sy thanks for reaching out! The inference time discrepancy may be influenced by factors such as hardware limitations or suboptimal INT8 calibration. I recommend ensuring your device supports efficient INT8 acceleration and calibrating the model with representative data. For further optimization, consider exporting to TensorRT with INT8 as it often provides greater speed-ups. You can find more details in the [Export documentation](https://docs.ultralytics.com/modes/export/).

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #18185](https://github.com/ultralytics/ultralytics/pull/18185) 🎉! Here's a summary of the key changes introduced in this PR:

### Key Updates:
- **Improved Argument Validation**: A new `validate_args` function has been added to ensure only supported arguments are passed during export, reducing the likelihood of errors.
- **Updated Export Format Configurations**: Specific configurations now support features like `half`, `int8`, and `batch`, improving clarity and usability.
- **Refined Workflows**: The export and benchmarking processes now leverage enhanced validations, making the process more robust and predictable.
- **Version Bump**: The library version has been updated from `8.3.52` to `8.3.53`.

These updates aim to enhance the export functionality, reduce errors, and improve overall usability. If this addresses the issue you were experiencing, we encourage you to test it out!

### How to Test:
1. Install the latest changes directly from the `main` branch:
   ```bash
   pip install git+https://github.com/ultralytics/ultralytics.git@main
   ```
2. OR wait until the next official PyPI release to test the updated version.

### Request for Feedback:
Please let us know if the changes in this PR resolve your issue. If the problem persists or you encounter any new challenges, feel free to provide additional feedback. 

Thank you for reporting this issue 🙏! Your input is invaluable in helping us improve. 🚀
